lib/iris/_data_manager.py

```diff
@@ -261,12 +287,26 @@ def dtype(self):
     @property
     def ndim(self):
         """The number of dimensions covered by the data being managed."""
-        return self.core_data().ndim
+        return len(self.shape)
 
     @property
     def shape(self):
         """The shape of the data being managed."""
-        return self.core_data().shape
+        if self.core_data() is None:
+            result = self._shape
+        else:
+            result = self.core_data().shape
+        return result
+
+    def is_dataless(self):
+        """Determine whether the cube is dataless.
```

Comment:
@ESadek-MO Perhaps it's best to not use `dataless` to describle `is_dataless` e.g., maybe something like `Determine whether the cube has no data.` instead?